### PR TITLE
sensor noise parameter control dialog

### PR DESCRIPTION
**Pull request by stmoon**
I implemented the sensor noise parameter control dialog for flight test and analysis.

https://youtu.be/5V67fofzN9s

The idea was inspired by @hamishwillee while add random walk noise for gps in this simulation. 
(refer to https://github.com/PX4/jMAVSim/pull/64)

I hope that it is useful to analyze and understand the px4 control code.
  
please check it @bkueng and @TSC21 

Thanks in advance.

SungTae Moon

### CODE REVIEW

**Comment by TSC21** [comment]
@stmoon is there any way you could actually integrate this with the UI of jMAVSim, instead of having a new window for it?

**Comment by stmoon** [comment]
If you think it is better than now, I will try it again.

**Comment by TSC21** [comment]
I think it will look better but if it is too much work you don't have to spend time on it. As is, is more than enough.

**Comment by stmoon** [comment]
Thanks @TSC21 

Actually, I think it spend some time. How about using the current version ?
(However I will have plan to insert the dialog to main UI for next time)

**Comment by TSC21** [comment]
Let's wait for @bkueng review.

**Comment by stmoon** [comment]
Thanks @TSC21

**Comment by stmoon** [comment]
Based on advice of @TSC21, I integrated the sensor parameter control panel with the main UI of jMAVSim (not dialog)

the shortcut key is same ('d' key)

Thanks for your advice of @TSC21  ;-)

**Comment by TSC21** [comment]
@stmoon can you share us a new video with it working? Thanks

**Comment by stmoon** [comment]
please check the video 

https://youtu.be/e3Tppt9nteM

Thanks.

File: src/me/drton/jmavsim/Visualizer3D.java [hunk 1]
```diff
@@ -1004,6 +1049,10 @@ public void keyReleased(KeyEvent e) {
                 case KeyEvent.VK_R :
                     toggleReportPanel();
                     break;
+
+                case KeyEvent.VK_D :
```
[the remark this answers is not in the document]
Comment (mpaperno) [reply src/me/drton/jmavsim/Visualizer3D.java]: Perhaps starting with the built-in key reference?  ;-)   https://github.com/PX4/jMAVSim/blob/master/src/me/drton/jmavsim/Simulator.java#L698

Great PR!  (Also 👍  adding the GPS noise!)

File: src/me/drton/jmavsim/SimpleSensors.java [hunk 2]
```diff
@@ -221,8 +221,53 @@ public void update(long t) {
...
+            gpsNoiseStdDev = value;
+        }
+        else {
+            System.out.printf("ERROR: unknown param");
+        }
+    }
+
+    @Override
+    public float param(String name) {
+        if ( name == "noise_Acc" ) {
```
[the remark this answers is not in the document]
Comment (stmoon) [reply src/me/drton/jmavsim/SimpleSensors.java]: Thanks.. I will change it

File: src/me/drton/jmavsim/Visualizer3D.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (stmoon) [reply src/me/drton/jmavsim/Visualizer3D.java]: I will write the key command in the README @bkueng 

Thanks and good advice ;-) @mpaperno

File: src/me/drton/jmavsim/Visualizer3D.java [hunk 3]
```diff
@@ -66,6 +68,7 @@
  
     
     private final World world;
+    private AbstractVehicle vehicle = null;
```
[the remark this answers is not in the document]
Comment (stmoon) [reply src/me/drton/jmavsim/Visualizer3D.java]: Good idea. @bkueng.

I will apply it.

Thanks

**Comment by stmoon** [comment]
@bkueng,  please check it

**Comment by stmoon** [comment]
Hi, @bkueng

I merged the updated master.

Could you check the modified codes? I want to go to the next step.

Thanks.

File: README.md [hunk 4]
```diff
@@ -65,6 +65,47 @@ On **Windows** use `;` instead of `:` in -cp:
 java -cp lib/*;out/production/jmavsim.jar me.drton.jmavsim.Simulator
 ```
 
+#### Key command ####
+
+Views:
+    F    - First-person-view camera.
```
Comment (bkueng) [inline README.md]: This does not properly render on github: have a look at the preview here: https://github.com/stmoon/jMAVSim/blob/402532912ff2562085852e2a3735b1a726d43057/README.md.
You can use a list for these, like this:
```
- F    - First-person-view camera.
```

File: README.md [hunk 5]
```diff
@@ -133,7 +174,7 @@ AbstractMulticopter vehicle = new Quadcopter(world, "models/3dr_arducopter_quad_
 
 Custom MAVLink protocols can be used, no any recompilation needed, just specify XML file instead of `custom.xml` here:
 ```
-MAVLinkSchema schema = new MAVLinkSchema("mavlink/message_definitions/common.xml");
+MAVLinkSchema schema = new MAVLinkSchema("mavlink/message_definitions/common.xml
```
Comment (bkueng) [inline README.md]: Why did you change this?

File: README.md (hunk 4 above)
Comment (stmoon) [reply README.md]: Thanks @bkueng 

I modify the README according to markdown syntax

File: README.md (hunk 5 above)
Comment (stmoon) [reply README.md]: Sorry, It's my mistake.. I changed it again.

**Comment by stmoon** [comment]
I'm analyzing  the unnecessary compile step which you say. 
It will need some time to check it...

**Comment by stmoon** [comment]
Hi, @bkueng 

It seems to be due to intellij javac2.  (The problem is happened when flightplot is compiled)

Unfortunately, I didn't find the solution to remove the warning message until now, 

If it is not critical to merge it, how about merging it first?

Of course, I will try to find the solution.

**Comment by bkueng** [comment]
Ok, I'm merging this, but maybe we should remove the javac2 again, and replacing the forms file with something else?

**Comment by stmoon** [comment]
Thanks @bkueng ,

Good idea!! 

I will remove javac2 and use javac again!!

Thanks!!
